opentelemetry/Cargo.toml

```diff
@@ -24,15 +24,15 @@ rustdoc-args = ["--cfg", "docsrs"]
 futures-core = { workspace = true, optional = true }
 futures-sink = { version = "0.3", optional = true }
 pin-project-lite = { workspace = true, optional = true }
-thiserror = { workspace = true, optional = true}
+thiserror = { workspace = true }
```

Comment:
We want to keep optional dependency on `thiserror`. As of now, it is used only when `trace` feature flag is enabled.

Reply:
`thiserror` is used in [`opentelemetry::propagation`](https://github.com/open-telemetry/opentelemetry-rust/blob/257377315d3949bff73d627b5deccd7f6a85376d/opentelemetry/src/propagation/mod.rs#L23) which is not feature gated. Building with `--no-default-features` fails. Therefore `thiserror` is required even without the feature `trace`.

`cargo check -p opentelemetry --no-default-features` fails.

Reply:
propagators are only used in the context of traces. See if we can move it include if only if `trace` flag is enabled here - https://github.com/open-telemetry/opentelemetry-rust/blob/257377315d3949bff73d627b5deccd7f6a85376d/opentelemetry/src/lib.rs#L267

Reply:
and also, you need to sign the CLA in order for this to be merged :)

Reply:
If it's fine for you I would prefer to not sign the CLA and leave the fix to you? :sweat_smile: Thanks for getting back so quickly!

Reply:
Sure. Thanks for the issue :)